### PR TITLE
feat(firefox): added firefox extension

### PR DESCRIPTION
**Pull request by crisboarna**
## Description

Adding Mozilla Firefox extension.

- Adds New Tab via search query to Google, searching Firefox tabs in session files and history.
- Adds Search Tab to search directly open tabs from session.
- Adds Search History to search directly from history.

Limitations (due to limited AppleScript support in Firefox):
- When searching open tabs, the session file is read and parsed to get the list of open tabs. This means that the list of open tabs will not be updated until the session file is updated. This is done by Firefox when it checkpoints itself or when Firefox is closed.
- Selecting an open tab will result in cycling through open tabs until desired tab. This is due to the fact that AppleScript does not support opening a specific tab in Firefox.

## Screencast
image
image
image
image

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi 👋 

Thanks for your contribution 💪 

I have now tested your extension and I have some feedback ready for you:

* Should this extension open things in Firefox or just this default browser?

* Search Tabs doesn't seems to work as expected

Raycast-evqYgd4V@2x


I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

File: .gitignore [hunk 1]
```diff
@@ -151,4 +151,5 @@ examples/**/*.zip
 extensions/**/*.zip
 
 # IntelliJ / WebStorm
-.idea/
\ No newline at end of file
+.idea/
+*.iml
```
Comment (pernielsentikaer) [inline .gitignore:155]:
```suggestion
```

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi 👋 

Thanks for your contribution 💪 

I have now tested your extension and I have some feedback ready for you:

* First metadata image doesn't add any value, could you remove that?
* Could you also use utlis package instead of sql.js here?

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

**Comment by pernielsentikaer** [comment]
Could you remove your .gitignore @crisboarna 😊

**Comment by crisboarna** [comment]
Hi @pernielsentikaer,

This PR can be reviewed now.

**Review by pernielsentikaer — APPROVED** [review]
Hi 👋 

Looks good to me, approved 🔥

**Pull request merged**